lib/utils/dom.js

```diff
@@ -412,3 +412,21 @@ function addStyle(css) {
 
 	return style;
 }
+
+export function waitForRemoval(el: HTMLElement): Promise<void> {
+	return new Promise(resolve => {
+		const observer = new MutationObserver(mutations => {
+			for (const mutation of mutations) {
+				for (const node of mutation.removedNodes) {
+					if (node === el || node.contains(el)) {
+						observer.disconnect();
+						resolve();
+						return;
+					}
+				}
+			}
+		});
+
+		observer.observe(document.body, { subtree: true, childList: true });
```

Comment:
Observing `document.body` would be expensive. Instead observing `ele.parentElement` would probably suffice, and then the `subtree` option can be removed.

Reply:
`parentElement` does not actually work.  This is because the JSAPI container is not removed when navigating between posts. I did try it before `document.body`.

Reply:
It am not uncertain why it would not work in this case. Since `target` is not moved (but is at all times the child of the same JSAPI container), it being manually removed in `userTagger` should trigger the observer when observing `target.parentElement`.

Reply:
`target.parentElement` does work, but the `MutationObserver` fires only once. Looking into why it does that.

Reply:
Update: `target.parentElement` issue could not be resolved. Listening on `document.body` is the only element that fires the MutationObserver correctly.

Reply:
@larsjohnsen Thoughts on what I could be missing?

Reply:
Using `target.parentElement` appears to work in the instances where the old tag icon is manually removed (#5087). However, when Reddit generates a new JSAPI container (which it seemingly does at random) only `document.body` will catch that.

Since we are stuck with using `document.body`, one thing that will make it less expensive would be to add the removal listener at a later stage, and potentially remove it when the popup is closed. The popup becomes visible here https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/f78d299d5840281dc92a5c8f48502166952f5c80/lib/modules/hover.js#L273